lucien-ios/LoginScreen/LoginViewModel.swift

```diff
@@ -0,0 +1,63 @@
+//
+//  LoginManager.swift
+//  lucien-ios
+//
+//  Created by Fang, Gracie on 9/29/17.
+//  Copyright © 2017 Intrepid Pursuits. All rights reserved.
+//
+
+final class LoginViewModel {
+
+    let lucienAPIClient = LucienAPIClient()
+    var currentUser: User?
+    var dashboardData: Dashboard?
+    var hasCollection: Bool = false
+
+    func getCurrentUser(completion: @escaping () -> Void) {
+        lucienAPIClient.getCurrentUser { response in
+            switch response {
+            case .success(let user):
+                self.currentUser = user
+                completion()
+            case .failure(let error):
+                print(error)
+            }
+        }
+    }
+
+    func authenticateUser(code: String, completion: @escaping () -> Void) {
+        lucienAPIClient.authenticateUser(code: code) { response in
+            switch response {
+            case .success(let authenticationResponse):
+                LucienRequest.authorizationHeader = "Bearer " + authenticationResponse.token
+                completion()
+            case .failure(let error):
+                print(error)
+            }
+        }
+    }
+
+    func getDashboard(completion: @escaping () -> Void) {
+        lucienAPIClient.getDashboard { response in
+            switch response {
+            case .success(let result):
+                self.dashboardData = result
+                completion()
+            case .failure(let error):
+                print(error)
+            }
+        }
+    }
+
+    func hasCollection(completion: @escaping () -> Void) {
```

Comment:
This completion isn't doing anything. The point of a completion is to act as a callback when an action completes, and to return any information needed from the action. In this case, the info we need from the action is `hasCollection` so the completion should be of type `(Bool) -> Void` and it should return the result on success. If you don't know why, I can clarify